kong/tools/dns.lua

```diff
@@ -14,20 +15,23 @@ local setup_client = function(conf)
   -- servers must be reformatted as name/port sub-arrays
   if conf.dns_resolver then
     for i, server in ipairs(conf.dns_resolver) do
-      local ip, port = server:match("^([^:]+)%:*(%d*)$")
-      servers[i] = { ip, tonumber(port) or 53 }   -- inserting port if omitted
+      local s = utils.normalize_ip(server)
+      servers[i] = { s.host, s.port or 53 }   -- inserting port if omitted
     end
   end
     
   local opts = {
     hosts = conf.dns_hostsfile,
     resolvConf = nil,                -- defaults to system resolv.conf
     nameservers = servers,           -- provided list or taken from resolv.conf
+    enable_ipv6 = true,              -- allow for ipv6 nameserver addresses
```

Comment:
Wanted to make the same comment but now consistently afraid of being called too pedantic or detail oriented :)